src/Umbraco.Core/Services/ContentService.cs

```diff
@@ -2575,8 +2575,18 @@ private Attempt<PublishStatus> SaveAndPublishDo(IContent content, int userId = 0
                         _publishingStrategy.PublishingFinalized(uow, descendants, false);
                     }
 
-                    Audit(uow, AuditType.Publish, "Save and Publish performed by user", userId, content.Id);
                     uow.Commit();
+
+                    if (publishStatus.StatusType == PublishStatusType.Success)
+                    {
+                        Audit(uow, AuditType.Publish, "Save and Publish performed by user", userId, content.Id);
+                    }
+                    else
+                    {
+                        Audit(uow, AuditType.Save, "Save performed by user", userId, content.Id);
+                    }
+                    uow.Commit();

```

Comment:
Thanks @lssweatherhead ! 
There's a second `uow.Commit` now, is that an oversight or on purpose?

Reply:
No problem @nul800sebastiaan :)

So the first `uow.Commit()` saves/publishes the content item and populates the content.id property. The second one saves the umbraco log data (with the appropriate content.id in place)